### PR TITLE
Enable ftrace only without `NDEBUG`, ftrace `EncodeTensorMapTiled`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Example print:
```C++
[ RUN      ] TMALdstTest/TMALdstTest.LoadCompleteTensor/4D_NoSwizzle_float
Entering EncodeTensorMapTiled(global_address=, 0x7ffd19e00000, global_dim=, 4 4 4 4, global_strides=, 16 64 256, box_dim=, 4 4 4 4, element_strides=, 1 1 1 1, data_type=, float, interleave=, NoInterleave, swizzle=, NoSwizzle, l2_promotion=, NoL2Promotion, oob_fill=, NoOOBFill, tensor_rank=, 4)
Leaving EncodeTensorMapTiled
[       OK ] TMALdstTest/TMALdstTest.LoadCompleteTensor/4D_NoSwizzle_float (299 ms)
```

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

File: tests/cpp/test_gpu_utils.cpp [hunk 1]
```diff
@@ -46,9 +47,13 @@ TEST_F(NVFuserTest, FunctionTrace1) {
       ss.str(),
       ::testing::HasSubstr("Leaving myFavoriteFunction returning 3 at "));
   EXPECT_THAT(ss.str(), ::testing::HasSubstr("test_gpu_utils.cpp:31"));
+#else
+  GTEST_SKIP() << "Test only runs in debug mode";
```
Comment (naoyam) [inline tests/cpp/test_gpu_utils.cpp:51]: This part, where `NDEBUG` is not defined, should be for the debug mode, right?

Comment (zasdfgbnm) [reply tests/cpp/test_gpu_utils.cpp:51]: Hmm, yes, my mind was flipped...

Comment (zasdfgbnm) [reply tests/cpp/test_gpu_utils.cpp:51]: Changed `#ifndef NDEBUG` to `#ifdef NDEBUG`

**Comment by zasdfgbnm** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM. Could you also correct the PR title? It should be "only without NDEBUG".

Any specific reason to disable with NDEBUG?

**Comment by zasdfgbnm** [comment]
> Any specific reason to disable with NDEBUG?

No strong reason. Just feeling that this is only useful for nvFuser developers, who usually build with debug. For a user, if this is not useful, better not to have it, so that we can more aggressively enable this feature on hot path.

**Pull request merged**